doc/README.md

```diff
@@ -0,0 +1,51 @@
+# Tokenlon
+
+Tokenlon is a decentralized exchanging protocol consists of multiple components with a single static entry. Users can trade assets with a quote from a professional market maker or swap with other AMM protocol with no husstle. Also, complex contract interactions can be simplified by leveraging the approval mechanism of the protocol. Tokenlon has its own native token named LON and the protocol is governed by LON token holders. Also, there're other functions in the token economic system which provide incentive for LON token holders and help keeping the protocol sustainable.
+
+There are three major categories of contracts in this repo:
+
+1. [Tokenlon Protocol Infrastructure](#Protocol-Infrastructure)
+
+2. [Executing Contracts](#Executing-Contracts)
+
+3. [LON Token Peripherals](#LON-Token-Peripherals)
+
+![image info](../tokenlon_architecture.png)
+
+# Protocol Infrastructure
+
+## UserProxy & Tokenlon
+
+User proxy is the entry of the whole procotol while Tokenlon is a transparent upgradeable proxy of it. User proxy navigate users to a specific strategy contract. Meanwhile, it has a multicall entry which allows batching calls between differnt strategy contracts in a single transaction.
```

Comment:
Strategy contracts are not mentioned later in the article? And
shall we explain the concept of strategy contract?

Reply:
Yeah I think how strategy contracts are integrated should be at least mentioned here. Will fix.